_inc/lib/plugins.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Plugins Library
+ *
+ * Helper functions for installing and activating plugins.
+ *
+ * Used by the REST API
+ *
+ * @autounit api plugins
+ */
+
+include_once( 'class.jetpack-automatic-install-skin.php' );
+
+class Jetpack_Plugins {
+
+	/**
+	 * Install and activate a plugin.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param string $slug Plugin slug.
+	 *
+	 * @return bool|WP_Error True if installation succeeded, error object otherwise.
+	 */
+	public static function install_and_activate_plugin( $slug ) {
+		// Check if get_plugins() function exists. This is required on the front end of the
+		// site, since it is in a file that is normally only loaded in the admin.
+		if ( ! function_exists( 'get_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_id = self::get_plugin_id_by_slug( $slug );
+
+		if ( ! $plugin_id ) {
+			$installed = self::install_plugin( $slug );
+			if ( is_wp_error( $installed ) ) {
+				return $installed;
+			}
+			$plugin_id = self::get_plugin_id_by_slug( $slug );
+			return activate_plugin( $plugin_id );
```

Comment:
Note that `activate_plugin` returns `null` on successful activation, so you'll want to update the logic here to return `true` on success (probably by checking if it's NOT a `WP_Error`). You might be able to benefit from the other review comments from this review: https://github.com/Automattic/jetpack/pull/8450#pullrequestreview-86841600.

Reply:
Oops, my bad. Oh those dreaded return values.